src/database.rs

```diff
@@ -195,6 +199,125 @@ async fn aggregate_stats(pool: &sqlx::PgPool) -> Result<()> {
     Ok(())
 }
 
+async fn aggregate_stats_by_context(pool: &sqlx::PgPool) -> Result<()> {
+    let _ = sqlx::query!(
+        r#"
+        INSERT INTO
+          aggregated_stats_by_context (
+            day,
+            server_context,
+            total_users,
+            total_nonbridged_users,
+            total_room_count,
+            daily_active_users,
+            daily_active_rooms,
+            daily_messages,
+            daily_sent_messages,
+            daily_active_e2ee_rooms,
+            daily_e2ee_messages,
+            daily_sent_e2ee_messages,
+            monthly_active_users,
+            r30_users_all,
+            r30_users_android,
+            r30_users_ios,
+            r30_users_electron,
+            r30_users_web,
+            r30v2_users_all,
+            r30v2_users_android,
+            r30v2_users_ios,
+            r30v2_users_electron,
+            r30v2_users_web,
+            daily_user_type_native,
+            daily_user_type_bridged,
+            daily_user_type_guest,
+            daily_active_homeservers
+          )
+        SELECT
+          local_timestamp:: DATE,
+          server_context,
+          SUM(total_users),
+          SUM(total_nonbridged_users),
+          SUM(total_room_count),
+          SUM(daily_active_users),
+          SUM(daily_active_rooms),
+          SUM(daily_messages),
+          SUM(daily_sent_messages),
+          SUM(daily_active_e2ee_rooms),
+          SUM(daily_e2ee_messages),
+          SUM(daily_sent_e2ee_messages),
+          SUM(monthly_active_users),
+          SUM(r30_users_all),
+          SUM(r30_users_android),
+          SUM(r30_users_ios),
+          SUM(r30_users_electron),
+          SUM(r30_users_web),
+          SUM(r30v2_users_all),
+          SUM(r30v2_users_android),
+          SUM(r30v2_users_ios),
+          SUM(r30v2_users_electron),
+          SUM(r30v2_users_web),
+          SUM(daily_user_type_native),
+          SUM(daily_user_type_bridged),
+          SUM(daily_user_type_guest),
+          COUNT(homeserver)
+        FROM
+          (
+            SELECT
+              DISTINCT ON (homeserver, local_timestamp:: DATE) *
+            FROM
+              reports
+            WHERE
+              local_timestamp:: DATE >= (
+                SELECT
+                  COALESCE(MAX(day), 'EPOCH':: DATE)
+                FROM
+                  aggregated_stats
+              )
+            ORDER BY
+              homeserver,
+              local_timestamp:: DATE,
+              local_timestamp DESC
+          ) as pg_sucks
+        WHERE server_context IS NOT NULL
```

Comment:
Not sure we need this. If all reports must provide context, then we better make this column `not null`, or we can just remove this line and have one row for "contextless" stats, which may or may not be fetched (added/summed) via `/get_aggregated_stats`

Reply:
This is needed for the current table schema, as we have a primary key constraint over `(day, server_context)` to be able to use `GROUP BY` in the way that we do. The way to solve this would be to plan the schema more properly, replacing null values with something else, combining all aggregated stats into one table and writing a separate query for summing all contexts if we want aggregated stats for everything.

Reply:
The one thing we can't change is existing reports, meaning we can't add a `not null` constraint to the context column in the reports table.

Reply:
But what if we just remove this exact line? It should just create a row per day where context is null. This way we would at least use `aggregated_stats_by_context.server_context` nullability property. And we wouldn't use any values from the reports

Reply:
Can we then make `aggregated_stats_by_context.server_context` `not null`?

Reply:
Ah, I see, primary key constraint implicitly constraints a column to `not null`